File: hw/tipantiza/u3/hw26TemplateMethodPattern/TemplateMethod_js/model/Coffee.js
const CaffeineBeverage = require('./CaffeineBeverage');
const readline = require('readline');

class Coffee extends CaffeineBeverage {
    brew() {
        console.log("Dripping coffee through filter");
    }

    addCondiments() {
        console.log("Adding sugar and milk");
    }

    async wantsCondiments() {
        const answer = await this.getUserInput();
        return answer.toLowerCase().startsWith('y');
    }

    getUserInput() {
        const rl = readline.createInterface({
            input: process.stdin,
            output: process.stdout
        });

        return new Promise((resolve) => {
            rl.question("Would you like milk and sugar with your coffee (y/n)? ", (answer) => {
                rl.close();
                resolve(answer);
            });
        });
    }
}

module.exports = Coffee;

File: hw/nacato/u3/hw25TemplateMethod/TemplateMethod.py/template_method/coffee.py
from caffeine_beverage import CaffeineBeverage


class Coffee(CaffeineBeverage):

    def brew(self):
        print("Dripping coffee through filter")

    def add_condiments(self):
        print("Adding sugar and milk")

    def get_user_input(self):
        return input("Would you like milk and sugar with your coffee (y/n)? ")